Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Midi/MidReader.cs

```diff
@@ -548,12 +588,46 @@ private static void ReadNotes(IList<MidiEvent> track, IList<MessageProcessParams
 
                     if (psEvent.value == MidIOHelper.SYSEX_VALUE_PHRASE_START)
                     {
-                        sysexEventQueue.Add(new PhaseShiftSysExStart(psEvent));
+                        // Check for duplicates
+                        // (a corresponding note off must be found, two note ons that are the same are disallowed)
+                        bool duplicate = false;
+                        foreach (var (queuedSysex, _) in sysexEventQueue)
```

Comment:
Try to avoid foreach iterator allocation within potentially high density loops like these.